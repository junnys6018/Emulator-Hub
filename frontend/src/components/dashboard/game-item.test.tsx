/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import renderer from 'react-test-renderer';
import GameItem, { GameCard, GameListItem } from './game-item';
import { MemoryRouter } from 'react-router';

const props = {
    image: '/image.png',
    gameUuid: 'mock',
    imageRendering: 'pixelated' as const,
    name: 'Zelda',
    onActiveCallback() {},
};

test('<GameCard /> snapshot', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <GameCard {...props} />
            </MemoryRouter>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('<GameListItem /> snapshot', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <GameListItem {...props} />
            </MemoryRouter>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('<GameItem /> snapshot', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <GameItem {...props} />
            </MemoryRouter>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
