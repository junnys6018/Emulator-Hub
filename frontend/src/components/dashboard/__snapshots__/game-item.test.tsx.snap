// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameCard /> snapshot 1`] = `
<div
  className="flex flex-col"
>
  <button
    className="select-none flex-grow rounded-2xl bg-cover bg-center filter drop-shadow hover:ring-2 hover:ring-green-500"
    style={
      Object {
        "aspectRatio": "1",
        "backgroundImage": "url(/image.png)",
        "imageRendering": "pixelated",
      }
    }
  />
  <div
    className="flex items-center mx-3 pt-1"
  >
    <span
      className="text-sm mr-auto"
    >
      Zelda
    </span>
    <button
      className="mr-2 hover:text-green-500 p-1"
    >
      <svg
        fill="currentColor"
        height="12px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
        />
      </svg>
    </button>
    <button
      className="hover:text-green-500 p-1"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="12px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<GameItem /> snapshot 1`] = `
Array [
  <div
    className="flex items-center md:hidden"
  >
    <button
      className="flex-shrink-0 rounded bg-cover bg-center mr-4"
      style={
        Object {
          "backgroundImage": "url(/image.png)",
          "height": "80px",
          "imageRendering": "pixelated",
          "width": "80px",
        }
      }
    />
    <span
      className="mr-auto truncate"
    >
      Zelda
    </span>
    <button
      className="ml-2 active:text-green-500"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 192 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        />
      </svg>
    </button>
  </div>,
  <div
    className="flex flex-col hidden md:flex"
  >
    <button
      className="select-none flex-grow rounded-2xl bg-cover bg-center filter drop-shadow hover:ring-2 hover:ring-green-500"
      style={
        Object {
          "aspectRatio": "1",
          "backgroundImage": "url(/image.png)",
          "imageRendering": "pixelated",
        }
      }
    />
    <div
      className="flex items-center mx-3 pt-1"
    >
      <span
        className="text-sm mr-auto"
      >
        Zelda
      </span>
      <button
        className="mr-2 hover:text-green-500 p-1"
      >
        <svg
          fill="currentColor"
          height="12px"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="12px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          />
        </svg>
      </button>
      <button
        className="hover:text-green-500 p-1"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="12px"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="12px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          />
        </svg>
      </button>
    </div>
  </div>,
]
`;

exports[`<GameListItem /> snapshot 1`] = `
<div
  className="flex items-center"
>
  <button
    className="flex-shrink-0 rounded bg-cover bg-center mr-4"
    style={
      Object {
        "backgroundImage": "url(/image.png)",
        "height": "80px",
        "imageRendering": "pixelated",
        "width": "80px",
      }
    }
  />
  <span
    className="mr-auto truncate"
  >
    Zelda
  </span>
  <button
    className="ml-2 active:text-green-500"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 192 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
      />
    </svg>
  </button>
</div>
`;
